### PR TITLE
Filter modal refactor

### PR DESCRIPTION
**Pull request by RobertJoonas**
### Changes

A refactoring PR that prepares for adding a new PropFilterModal component.

### Tests
- [x] This PR does not require tests

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: assets/js/dashboard/stats/modals/regular-filter-modal.js [hunk 1]
```diff
@@ -35,28 +33,22 @@ function getFormState(filterGroup, query) {
...
+class RegularFilterModal extends React.Component {
   constructor(props) {
     super(props)
     const query = parseQuery(props.location.search, props.site)
-    const selectedFilterGroup = this.props.match.params.field || 'page'
-    const formState = getFormState(selectedFilterGroup, query)
+    const formState = getFormState(props.filterGroup, query)
 
-    this.state = { selectedFilterGroup, query, formState }
+    this.state = { query, formState }
```
Comment (ukutaht) [inline assets/js/dashboard/stats/modals/regular-filter-modal.js:51]: Good call - the filter group doesn't need to be in `state`.

**Review by ukutaht — APPROVED** [review]
Looks good 👍

As I understand the code changes, there are no changes to actual behaviour at all?

**Review by vinibrsl — APPROVED** [review]
Looks good! One small suggestion inline :)

File: assets/js/dashboard/components/filter-type-selector.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    return (
+      shouldDisplay && (
+        <Menu.Item>
+          {({ active }) => (
+            <span
+              onClick={() => props.onSelect(type)}
+              className={classNames(
+                active ? "bg-gray-100 dark:bg-gray-900 text-gray-900 dark:text-gray-100" : "text-gray-700 dark:text-gray-200",
+                "cursor-pointer block px-4 py-2 text-sm"
+              )}
```
[the remark this answers is not in the document]
Comment (RobertJoonas) [reply assets/js/dashboard/components/filter-type-selector.js:22]: Good catch, thanks! :)

[the remark this answers is not in the document]
Comment (RobertJoonas) [reply assets/js/dashboard/components/filter-type-selector.js]: https://github.com/plausible/analytics/pull/2806/commits/c2ccdca232a4615569d4338daab6acb6e6c29734

**Pull request merged**